include/nbl/asset/CGLTFPipelineMetadata.h

```diff
@@ -0,0 +1,74 @@
+#ifndef __NBL_C_GLTF_PIPELINE_METADATA_H_INCLUDED__
+#define __NBL_C_GLTF_PIPELINE_METADATA_H_INCLUDED__
+
+#include "nbl/asset/metadata/IAssetMetadata.h"
+#include "nbl/asset/metadata/IRenderpassIndependentPipelineMetadata.h"
+#include "nbl/asset/ICPUDescriptorSet.h"
+#include "nbl/asset/ICPUPipelineLayout.h"
+
+namespace nbl
+{
+    namespace asset
+    {
+        // TODO pack push constants as a struct to CGLTFPipelineMetadata since it will have to many bytes above 128b
+
+        class CGLTFPipelineMetadata final : public IAssetMetadata, public IRenderpassIndependentPipelineMetadata
+        {
+        public:
+
+            CGLTFPipelineMetadata() {}
+            virtual ~CGLTFPipelineMetadata() {}
+
+            enum E_ALPHA_MODE : uint32_t
+            {
+                EAM_OPAQUE = core::createBitmask({0}),
+                EAM_MASK = core::createBitmask({1}),
+                EAM_BLEND = core::createBitmask({2})
+            };
+
+            #include "nbl/nblpack.h"
+            //! This struct is compliant with GLSL's std140 and std430 layouts
+            struct alignas(16) SGLTFMaterialParameters
+            {
+                struct SPBRMetallicRoughness
+                {
+                    core::vector4df_SIMD baseColorFactor = core::vector4df_SIMD(1.f, 1.f, 1.f, 1.f);
+                    float metallicFactor = 1.f;
+                    float roughnessFactor = 1.f;
+                };
```

Comment:
you already have these in pushconstants

metadata is for stuff that's not in push constants